processing/src/main/java/io/druid/query/groupby/epinephelinae/BufferGrouper.java

```diff
@@ -428,10 +431,17 @@ private void growIfPossible()
     final ByteBuffer entryBuffer = tableBuffer.duplicate();
     final ByteBuffer keyBuffer = tableBuffer.duplicate();
 
+    List<Map<Integer, Integer>> reloationMaps = new ArrayList<>(aggregators.length);
+    for (int i = 0; i < aggregators.length ; i++)
+    {
+      reloationMaps.add(new HashMap<Integer, Integer>(buckets));
```

Comment:
At least in groupby, the query code will always be growing by relocating data from the old tableBuffer to the new tableBuffer, so collisions shouldn't be an issue.